src/NServiceBus.RabbitMQ.AcceptanceTests/NServiceBus.RabbitMQ.AcceptanceTests.csproj

```diff
@@ -21,6 +21,7 @@
     <DefineConstants>DEBUG;TRACE</DefineConstants>
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
```

Comment:
If that's the overall consensus then I'll go with it, but I am opposed to it. As I said, I tried this in other OSS projects and the cost of friction for contributors outweighed the benefit.

For closed source projects I would always set this flag.